src/vs/vscode.d.ts

```diff
@@ -4719,6 +4719,170 @@ declare module 'vscode' {
 		resolveTask(task: Task, token?: CancellationToken): ProviderResult<Task>;
 	}
 
+	/**
+	 * Content settings for a webview.
+	 */
+	export interface WebviewOptions {
+		/**
+		 * Should scripts be enabled in the webview content?
+		 *
+		 * Defaults to false (scripts-disabled).
+		 */
+		readonly enableScripts?: boolean;
+
+		/**
+		 * Should command uris be enabled in webview content?
+		 *
+		 * Defaults to false.
+		 */
+		readonly enableCommandUris?: boolean;
+
+		/**
+		 * Root paths from which the webview can load local (filesystem) resources using the `vscode-resource:` scheme.
+		 *
+		 * Default to the root folders of the current workspace plus the extension's install directory.
+		 *
+		 * Pass in an empty array to disallow access to any local resources.
+		 */
+		readonly localResourceRoots?: ReadonlyArray<Uri>;
+	}
+
+	/**
+	 * A webview displays html content, like an iframe.
+	 */
+	export interface Webview {
+		/**
+		 * Content settings for the webview.
+		 */
+		readonly options: WebviewOptions;
+
+		/**
+		 * Title of the webview shown in UI.
+		 */
+		title: string;
```

Comment:
Not sure if this is a panel or widget property...

Reply:
It was a panel property but I moved it to the webview. The intention is that title is a common property pretty much anywhere the webview would be used. The owner of a webview (such as the `WebviewPanel`) just figures out how to display it

Reply:
> The intention is that title is a common property pretty much anywhere the webview would be used.

Could we say the same of dispose? Somehow I was missing it from the panel and I would also miss that from something like `WebviewWindow`...

Reply:
Yes I think most conceivable future api uses of the `Webview` would also want a `dispose` method. I'll move `dispose` and `onDidDispose` from the `WebviewPanel` class to the `Webview` itself

Reply:
Yeah, but that's the opposite of what I wanted ;-) It's kinda bogus to say `createWebviewPanel` but to then call `webviewPanel.webview.dispose()`.

Reply:
So you would prefer it on both objects?